### ISSUE TITLE
Sometimes IOErrors when piping to aplay

### ISSUE THREAD

**Issue by dbaumgarten** [lukasjapan/bt-speaker#30]
Sometimes it happens that despite successfully connecting no sound is playing. Looking at the logs I see:

```
Traceback (most recent call last):
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:   File "/usr/lib/python2.7/dist-packages/gi/override
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:     func_fdtransform = lambda _, cond, *data: callba
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:   File "/opt/bt-speaker/bt_manager/audio.py", line 1
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:     self.user_cb(self.user_arg)
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:   File "/opt/bt-speaker/bt_manager/audio.py", line 5
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:     self.raw_audio(self.read_transport())
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:   File "/opt/bt-speaker/bt_speaker.py", line 64, in 
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]:     # pipe to the play command
Jan 26 11:21:02 raspberrypi bt_speaker.py[1859]: IOError: [Errno 32] Broken pipe
```
Reconnecting does not help.
Restarting bt_speaker fixes this.
This seems to be happening mostly over night. At evening before everything works great and the next morning it doesn't. But it doesn't happen every night.

I don't no what exactly is causing the IOError, but I think it would be good to handle it gracefully.
For me just doing sys.exit(1) on such an error would be enough, as systemd will immediately restart the service and everything will be working again.

I am running bt-speaker on a Raspberry Pi 1B. 
(I know this is not the officially supported Hardware, but it is working really great. Much better than the pulseaudio solution I tried before.)

**Comment by dbaumgarten** [comment]
So, apparently the aplay-process sometimes crashes, which leads to an IOError whenever bt_speaker tries to pipe audio to it. 
I have no idea why aplay crashes this often (~once per day).
Just letting bt_speaker restart aplay on an IOError seems to solve the problem. I will open a PR for this.

**Comment by dbaumgarten** [comment]
Fixed with #37
